### PR TITLE
SRO: Add traces to sheets

### LINKED ISSUE #2554: Add Character traces to talent sheet UI
How do we represent the "Trace constellation/tree" in our current UI? how does the nesting work? can we parse datamine for it?
Can we making ui-sheets for the traces, and let it nest under the main basic/skill/ult/talent categories?

Blocks https://github.com/users/frzyc/projects/2/views/1?pane=issue&itemId=88957733

### PR DESCRIPTION
**Pull request by frzyc**
## Describe your changes
Add bonus ability and stat boost traces to sheets.
- Make bonus ability/ stat boost fade out if its not enabled (by ascension/level), and toggleable
- Add some utility functions to sheets

## Issue or discord link

- Resolves #2554

## Testing/validation
image

## Checklist before requesting a review (leave this PR as draft if any part of this list is not done.)

- [ ] I have commented my code in hard-to understand areas.
- [ ] I have made corresponding changes to README or wiki.
- [ ] For front-end changes, I have updated the corresponding English translations.
- [ ] I have run `yarn run mini-ci` locally to validate format and lint.
- [ ] If I have added a new library or app, I have updated the deployment scripts to ignore changes as needed

## Summary by CodeRabbit

- **New Features**
	- Added support for character sheets in multiple languages, enhancing localization with new JSON files for various languages including Chinese, German, Spanish, French, Indonesian, Japanese, Korean, Portuguese, Russian, Thai, and Vietnamese.
	- Introduced new functionality for managing character stat boosts and bonus abilities, improving character customization options.
  
- **Improvements**
	- Enhanced formatting of ability descriptions across multiple characters for better clarity and consistency.
	- Updated placeholder formats in ability descriptions to standardize representation.
	- Expanded the handling of character talents and eidolons, allowing for more dynamic rendering of components.

- **Bug Fixes**
	- Corrected formatting issues in ability descriptions to ensure accurate display of dynamic values.

- **Chores**
	- Removed outdated localization files that are no longer in use.

### CODE REVIEW

File: libs/sr/formula-ui/src/char/StatBoostBonusAbility.tsx [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+      <Translate ns="charSheet_gen" key18="req_lvl" values={{ 1: level }} />
+    ),
+    onClickable: (calc: Calculator) => calc.compute(own.char.lvl).val >= level,
+    disabled: (calc: Calculator) =>
+      calc.compute(own.char.lvl).val < level ||
+      !calc.compute(own.char[talentKey]).val,
+  }
+}
+
+export const bonusStatsReqMap: Record<
```
Comment (nguyentvan7) [inline libs/sr/formula-ui/src/char/StatBoostBonusAbility.tsx:45]: Don't know if you wanna add a TODO for other dependencies (e.g. bonus stat X might rely on bonus stat X-3 being unlocked, and it changes per-path)

Comment (nguyentvan7) [reply libs/sr/formula-ui/src/char/StatBoostBonusAbility.tsx:45]: It is datamineable (#2568)

File: libs/sr/stats/src/char.ts [hunk 2]
```diff
@@ -19,7 +19,16 @@ export function getCharStat(ck: CharacterGenderedKey) {
...
   ck: CharacterGenderedKey,
-  skType: 'basic' | 'skill' | 'ult' | 'talent' | 'technique' | 'eidolon',
+  skType:
+    | 'basic'
+    | 'skill'
+    | 'ult'
+    | 'talent'
+    | 'technique'
+    | 'eidolon'
+    | 'bonusAbility1'
```
Comment (nguyentvan7) [inline libs/sr/stats/src/char.ts:29]: Seems possibly broken for bonus ability
image

**Review by nguyentvan7 — APPROVED** [review]

File: libs/sr/stats/src/char.ts (hunk 2 above)
Comment (frzyc) [reply libs/sr/stats/src/char.ts:29]: Yeah, I noticed this. I added a `// FIXME`
https://github.com/frzyc/genshin-optimizer/blob/67fe5671582edf4f0c6e165038f49c128875398e/libs/sr/formula-ui/src/char/sheetUtil.tsx#L76

Comment (frzyc) [reply libs/sr/stats/src/char.ts:29]: I did not touch datamine stuff so I did not dig too deeply.

**Pull request merged**